catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/FormulaEditorEditTextTest.java

```diff
@@ -0,0 +1,407 @@
+				.perform(click());
+		onView(withText(containsString(random)))
+				.perform(click());
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(click());
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(click());
+
+		onView(withId(R.id.formula_editor_keyboard_function))
+				.perform(click());
+
+		onView(allOf(withText(containsString(random)), withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
+				.perform(click());
+
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(click());
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(click());
+		onFormulaEditor()
+				.checkShows(" ");
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+	@Test
+	public void testNumberInsertion() {
+
+		onFormulaEditor()
+				.performEnterFormula(".1");
+		onFormulaEditor()
+				.checkShows("0" + decimalMark + "1 ");
+		setAbsoluteCursorPosition(0);
+		onFormulaEditor()
+				.performEnterFormula("5");
+		setAbsoluteCursorPosition(3);
+		onFormulaEditor()
+				.performEnterFormula("6");
+		onFormulaEditor()
+				.checkShows("50" + decimalMark + "61 ");
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+	@Test
+	public void testGoBackToDiscardChanges() {
+
+		onFormulaEditor()
+				.performEnterFormula("99.9+");
+		pressBack();
+		onView(withText(getCurrentActivity().getString(R.string.formula_editor_discard_changes_dialog_title)))
+				.check(matches(isDisplayed()));
+		onView(withText(no))
+				.perform(click());
+		checkToastMessage(R.string.formula_editor_changes_discarded);
+
+		onBrickAtPosition(1)
+				.checkShowsText("0 ");
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+	@Test
+	public void testDiscardDialog() {
+		onFormulaEditor()
+				.performEnterFormula("99");
+		pressBack();
+		onView(withText(getCurrentActivity().getString(R.string.formula_editor_discard_changes_dialog_title)))
+				.check(matches(isDisplayed()));
+		pressBack();
+
+		onFormulaEditor()
+				.performEnterFormula(".9");
+		pressBack();
+		onView(withText(getCurrentActivity().getString(R.string.formula_editor_discard_changes_dialog_title)))
+				.check(matches(isDisplayed()));
+		onView(withText(yes))
+				.perform(click());
+		checkToastMessage(R.string.formula_editor_changes_saved);
+
+		onView(allOf(withId(R.id.brick_change_size_by_edit_text), withEffectiveVisibility(Visibility.VISIBLE)))
+				.check(matches(withText("99" + decimalMark + "9 ")));
+		onView(allOf(withId(R.id.brick_change_size_by_edit_text), withEffectiveVisibility(Visibility.VISIBLE)))
+				.perform(click());
+		pressBack();
+
+		onView(allOf(withId(R.id.brick_change_size_by_edit_text), withEffectiveVisibility(Visibility.VISIBLE)))
+				.check(matches(withText("99" + decimalMark + "9 ")));
+
+		onView(allOf(withId(R.id.brick_change_size_by_edit_text), withEffectiveVisibility(Visibility.VISIBLE)))
+				.perform(click());
+		onView(withId(R.id.formula_editor_keyboard_delete))
```

Comment:
please use only wrapper methods of the formula editor wrapper instead of using ids directly.

formula editor will be replaced / refactored, so if all the ids change it would be nice that we could only update a single wrapper instead of a s*load of tests that reference every single button on it...